src/osp/Resource/Package.h

```diff
@@ -175,17 +193,30 @@ DependRes<TYPE_T> Package::add(std::string_view path,
 
     GroupType<TYPE_T> &group = entt::any_cast< GroupType<TYPE_T>& >(groupAny);
 
+    // Check if resource already exists
+    auto resIt = group.m_resources.find(path); // Find resource by path
+
+    if (resIt != group.m_resources.end())
+    {
+        // Resource already exists, replace if not loaded
+        if (!resIt->second.m_data.has_value())
+        {
+            resIt->second.m_data.emplace(std::forward<ARGS_T>(args)...);
+        }
+        return {};
+    }
+
     // Emplace without needing copy constructor
-    auto final = group.m_resources
-        .try_emplace(std::string{path}, false, std::forward<ARGS_T>(args)...);
+    auto newIt = group.m_resources
+        .try_emplace(std::string{path}, construct_tag(), std::forward<ARGS_T>(args)...);
```

Comment:
It may not make sense to have the parameter to the function Package::add take a std::string_view if you're always going to allocate a new string when you call try_emplace.